src/dmd/frontend.d

```diff
@@ -45,6 +46,8 @@ void initDMD()
     else
         static assert(0, "OS not supported yet.");
 
+    addDefaultVersionIdentifiers();
```

Comment:
You can now remove the OS detection above :)
I had to make a copy of this as `addDefaultVersionIdentifiers` was private and I wanted to keep the changes to a minimum in the initial version.